--- ssam-1.0.2/ssam/__init__.py.orig	2023-05-05 12:17:16.896905000 +0100
+++ ssam-1.0.2/ssam/__init__.py	2023-05-05 12:17:58.748199689 +0100
@@ -115,7 +115,7 @@
             df.to_feather(ifn, version=1)
         else:
             df.to_feather(ifn)
-        rcmd = 'library(feather); library(sctransform); mat <- t(as.matrix(read_feather("{0}"))); colnames(mat) <- 1:ncol(mat); res <- vst(mat{1}, return_gene_attr=TRUE, return_cell_attr=TRUE); write_feather(as.data.frame(t(res$y)), "{2}"); write_feather(as.data.frame(res$model_pars_fit), "{3}");'.format(ifn, vst_opt_str, ofn, pfn)
+        rcmd = 'library(arrow); library(sctransform); mat <- t(as.matrix(read_feather("{0}"))); colnames(mat) <- 1:ncol(mat); res <- vst(mat{1}, return_gene_attr=TRUE, return_cell_attr=TRUE); write_feather(as.data.frame(t(res$y)), "{2}"); write_feather(as.data.frame(res$model_pars_fit), "{3}");'.format(ifn, vst_opt_str, ofn, pfn)
         if plot_model_pars:
             plot_path = os.path.join(tmpdirname, 'model_pars.png')
             rcmd += 'png(file="%s", width=3600, height=1200, res=300); plot_model_pars(res, show_var=TRUE); dev.off();'%plot_path
