--- pysharc/sharc/pysharc/constants.py.old	2025-10-23 14:14:16.000000000 +0100
+++ pysharc/sharc/pysharc/constants.py	2026-05-13 10:36:26.645886000 +0100
@@ -113,3 +113,323 @@
     'Y': 1,
     'Z': 2
 }
+
+# au2a
+
+au2a = 0.529177211
+
+# au2eV
+
+au2eV = 27.2113987622
+
+# au2rcm
+
+au2rcm = 219474.631370e0
+
+# au2fs
+
+au2fs = 0.024188843e0
+
+# D2au
+
+D2au = 0.393430307
+
+# rcm_to_Eh
+
+rcm_to_Eh = 4.556335e-6
+
+# AMBERVEL_TO_AU
+
+AMBERVEL_TO_AU = 0.0009350161
+
+# BOHR_TO_ANG
+
+BOHR_TO_ANG = 0.529177211
+
+# ANG_TO_BOHR
+
+ANG_TO_BOHR = 1. / 0.529177211    # 1.889725989    # conversion from Angstrom to bohr
+
+# U_TO_AMU
+
+U_TO_AMU = 1. / 5.4857990943e-4
+
+# CM_TO_HARTREE
+
+CM_TO_HARTREE = 1. / 219474.6
+
+# HARTREE_TO_EV
+
+HARTREE_TO_EV = 27.211396132    # conversion factor from Hartree to eV
+
+# n_avogadro
+
+n_avogadro = 6.0221408e34
+
+# NUMBERS
+
+NUMBERS = {'H': 1, 'He': 2,
+           'Li': 3, 'Be': 4, 'B': 5, 'C': 6, 'N': 7, 'O': 8, 'F': 9, 'Ne': 10,
+           'Na': 11, 'Mg': 12, 'Al': 13, 'Si': 14, 'P': 15, 'S': 16, 'Cl': 17, 'Ar': 18,
+           'K': 19, 'Ca': 20,
+           'Sc': 21, 'Ti': 22, 'V': 23, 'Cr': 24, 'Mn': 25, 'Fe': 26, 'Co': 27, 'Ni': 28, 'Cu': 29, 'Zn': 30,
+           'Ga': 31, 'Ge': 32, 'As': 33, 'Se': 34, 'Br': 35, 'Kr': 36,
+           'Rb': 37, 'Sr': 38,
+           'Y': 39, 'Zr': 40, 'Nb': 41, 'Mo': 42, 'Tc': 43, 'Ru': 44, 'Rh': 45, 'Pd': 46, 'Ag': 47, 'Cd': 48,
+           'In': 49, 'Sn': 50, 'Sb': 51, 'Te': 52, 'I': 53, 'Xe': 54,
+           'Cs': 55, 'Ba': 56,
+           'La': 57,
+           'Ce': 58, 'Pr': 59, 'Nd': 60, 'Pm': 61, 'Sm': 62, 'Eu': 63, 'Gd': 64, 'Tb': 65, 'Dy': 66, 'Ho': 67, 'Er': 68, 'Tm': 69, 'Yb': 70, 'Lu': 71,
+           'Hf': 72, 'Ta': 73, 'W': 74, 'Re': 75, 'Os': 76, 'Ir': 77, 'Pt': 78, 'Au': 79, 'Hg': 80,
+           'Tl': 81, 'Pb': 82, 'Bi': 83, 'Po': 84, 'At': 85, 'Rn': 86,
+           'Fr': 87, 'Ra': 88,
+           'Ac': 89,
+           'Th': 90, 'Pa': 91, 'U': 92, 'Np': 93, 'Pu': 94, 'Am': 95, 'Cm': 96, 'Bk': 97, 'Cf': 98, 'Es': 99, 'Fm': 100, 'Md': 101, 'No': 102, 'Lr': 103,
+           'Rf': 104, 'Db': 105, 'Sg': 106, 'Bh': 107, 'Hs': 108, 'Mt': 109, 'Ds': 110, 'Rg': 111, 'Cn': 112,
+           'Nh': 113, 'Fl': 114, 'Mc': 115, 'Lv': 116, 'Ts': 117, 'Og': 118
+}
+
+# MASSES
+
+MASSES = {'H': 1.007825 * U_TO_AMU,
+          'He': 4.002603 * U_TO_AMU,
+          'Li': 7.016004 * U_TO_AMU,
+          'Be': 9.012182 * U_TO_AMU,
+          'B': 11.009305 * U_TO_AMU,
+          'C': 12.000000 * U_TO_AMU,
+          'N': 14.003074 * U_TO_AMU,
+          'O': 15.994915 * U_TO_AMU,
+          'F': 18.998403 * U_TO_AMU,
+          'Ne': 19.992440 * U_TO_AMU,
+          'Na': 22.989770 * U_TO_AMU,
+          'Mg': 23.985042 * U_TO_AMU,
+          'Al': 26.981538 * U_TO_AMU,
+          'Si': 27.976927 * U_TO_AMU,
+          'P': 30.973762 * U_TO_AMU,
+          'S': 31.972071 * U_TO_AMU,
+          'Cl': 34.968853 * U_TO_AMU,
+          'Ar': 39.962383 * U_TO_AMU,
+          'K': 38.963707 * U_TO_AMU,
+          'Ca': 39.962591 * U_TO_AMU,
+          'Sc': 44.955910 * U_TO_AMU,
+          'Ti': 47.947947 * U_TO_AMU,
+          'V': 50.943964 * U_TO_AMU,
+          'Cr': 51.940512 * U_TO_AMU,
+          'Mn': 54.938050 * U_TO_AMU,
+          'Fe': 55.934942 * U_TO_AMU,
+          'Co': 58.933200 * U_TO_AMU,
+          'Ni': 57.935348 * U_TO_AMU,
+          'Cu': 62.929601 * U_TO_AMU,
+          'Zn': 63.929147 * U_TO_AMU,
+          'Ga': 68.925581 * U_TO_AMU,
+          'Ge': 73.921178 * U_TO_AMU,
+          'As': 74.921596 * U_TO_AMU,
+          'Se': 79.916522 * U_TO_AMU,
+          'Br': 78.918338 * U_TO_AMU,
+          'Kr': 83.911507 * U_TO_AMU,
+          'Rb': 84.911789 * U_TO_AMU,
+          'Sr': 87.905614 * U_TO_AMU,
+          'Y': 88.905848 * U_TO_AMU,
+          'Zr': 89.904704 * U_TO_AMU,
+          'Nb': 92.906378 * U_TO_AMU,
+          'Mo': 97.905408 * U_TO_AMU,
+          'Tc': 98.907216 * U_TO_AMU,
+          'Ru': 101.904350 * U_TO_AMU,
+          'Rh': 102.905504 * U_TO_AMU,
+          'Pd': 105.903483 * U_TO_AMU,
+          'Ag': 106.905093 * U_TO_AMU,
+          'Cd': 113.903358 * U_TO_AMU,
+          'In': 114.903878 * U_TO_AMU,
+          'Sn': 119.902197 * U_TO_AMU,
+          'Sb': 120.903818 * U_TO_AMU,
+          'Te': 129.906223 * U_TO_AMU,
+          'I': 126.904468 * U_TO_AMU,
+          'Xe': 131.904154 * U_TO_AMU,
+          'Cs': 132.905447 * U_TO_AMU,
+          'Ba': 137.905241 * U_TO_AMU,
+          'La': 138.906348 * U_TO_AMU,
+          'Ce': 139.905435 * U_TO_AMU,
+          'Pr': 140.907648 * U_TO_AMU,
+          'Nd': 141.907719 * U_TO_AMU,
+          'Pm': 144.912744 * U_TO_AMU,
+          'Sm': 151.919729 * U_TO_AMU,
+          'Eu': 152.921227 * U_TO_AMU,
+          'Gd': 157.924101 * U_TO_AMU,
+          'Tb': 158.925343 * U_TO_AMU,
+          'Dy': 163.929171 * U_TO_AMU,
+          'Ho': 164.930319 * U_TO_AMU,
+          'Er': 165.930290 * U_TO_AMU,
+          'Tm': 168.934211 * U_TO_AMU,
+          'Yb': 173.938858 * U_TO_AMU,
+          'Lu': 174.940768 * U_TO_AMU,
+          'Hf': 179.946549 * U_TO_AMU,
+          'Ta': 180.947996 * U_TO_AMU,
+          'W': 183.950933 * U_TO_AMU,
+          'Re': 186.955751 * U_TO_AMU,
+          'Os': 191.961479 * U_TO_AMU,
+          'Ir': 192.962924 * U_TO_AMU,
+          'Pt': 194.964774 * U_TO_AMU,
+          'Au': 196.966552 * U_TO_AMU,
+          'Hg': 201.970626 * U_TO_AMU,
+          'Tl': 204.974412 * U_TO_AMU,
+          'Pb': 207.976636 * U_TO_AMU,
+          'Bi': 208.980383 * U_TO_AMU,
+          'Po': 208.982416 * U_TO_AMU,
+          'At': 209.987131 * U_TO_AMU,
+          'Rn': 222.017570 * U_TO_AMU,
+          'Fr': 223.019731 * U_TO_AMU,
+          'Ra': 226.025403 * U_TO_AMU,
+          'Ac': 227.027747 * U_TO_AMU,
+          'Th': 232.038050 * U_TO_AMU,
+          'Pa': 231.035879 * U_TO_AMU,
+          'U': 238.050783 * U_TO_AMU,
+          'Np': 237.048167 * U_TO_AMU,
+          'Pu': 244.064198 * U_TO_AMU,
+          'Am': 243.061373 * U_TO_AMU,
+          'Cm': 247.070347 * U_TO_AMU,
+          'Bk': 247.070299 * U_TO_AMU,
+          'Cf': 251.079580 * U_TO_AMU,
+          'Es': 252.082972 * U_TO_AMU,
+          'Fm': 257.095099 * U_TO_AMU,
+          'Md': 258.098425 * U_TO_AMU,
+          'No': 259.101024 * U_TO_AMU,
+          'Lr': 262.109692 * U_TO_AMU,
+          'Rf': 267. * U_TO_AMU,
+          'Db': 268. * U_TO_AMU,
+          'Sg': 269. * U_TO_AMU,
+          'Bh': 270. * U_TO_AMU,
+          'Hs': 270. * U_TO_AMU,
+          'Mt': 278. * U_TO_AMU,
+          'Ds': 281. * U_TO_AMU,
+          'Rg': 282. * U_TO_AMU,
+          'Cn': 285. * U_TO_AMU,
+          'Nh': 286. * U_TO_AMU,
+          'Fl': 289. * U_TO_AMU,
+          'Mc': 290. * U_TO_AMU,
+          'Lv': 293. * U_TO_AMU,
+          'Ts': 294. * U_TO_AMU,
+          'Og': 294. * U_TO_AMU
+}
+
+# ISOTOPES
+
+ISOTOPES = {'H': 'H-1',
+            'He': 'He-4',
+            'Li': 'Li-7',
+            'Be': 'Be-9*',
+            'B': 'B_11',
+            'C': 'C-12',
+            'N': 'N-14',
+            'O': 'O-16',
+            'F': 'F-19*',
+            'Ne': 'Ne-20',
+            'Na': 'Na-23*',
+            'Mg': 'Mg-24',
+            'Al': 'Al-27*',
+            'Si': 'Si-28',
+            'P': 'P-31*',
+            'S': 'S-32',
+            'Cl': 'Cl-35',
+            'Ar': 'Ar-40',
+            'K': 'K-39',
+            'Ca': 'Ca-40',
+            'Sc': 'Sc-45*',
+            'Ti': 'Ti-48',
+            'V': 'V-51',
+            'Cr': 'Cr-52',
+            'Mn': 'Mn-55*',
+            'Fe': 'Fe-56',
+            'Co': 'Co-59*',
+            'Ni': 'Ni-58',
+            'Cu': 'Cu-63',
+            'Zn': 'Zn-64',
+            'Ga': 'Ga-69',
+            'Ge': 'Ge-74',
+            'As': 'As-75*',
+            'Se': 'Se-80',
+            'Br': 'Br-79',
+            'Kr': 'Kr-84',
+            'Rb': 'Rb-85',
+            'Sr': 'Sr-88',
+            'Y': 'Y-89*',
+            'Zr': 'Zr-90',
+            'Nb': 'Nb-93*',
+            'Mo': 'Mo-98',
+            'Tc': 'Tc-98',
+            'Ru': 'Ru-102',
+            'Rh': 'Rh-103*',
+            'Pd': 'Pd-106',
+            'Ag': 'Ag-107',
+            'Cd': 'Cd-114',
+            'In': 'In-115',
+            'Sn': 'Sn-120',
+            'Sb': 'Sb-121',
+            'Te': 'Te-130',
+            'I': 'I-127*',
+            'Xe': 'Xe-132',
+            'Cs': 'Cs-133*',
+            'Ba': 'Ba-138',
+            'La': 'La-139',
+            'Ce': 'Ce-140',
+            'Pr': 'Pr-141*',
+            'Nd': 'Nd-142',
+            'Pm': 'Pm-145',
+            'Sm': 'Sm-152',
+            'Eu': 'Eu-153',
+            'Gd': 'Gd-158',
+            'Tb': 'Tb-159*',
+            'Dy': 'Dy-164',
+            'Ho': 'Ho-165*',
+            'Er': 'Er-166',
+            'Tm': 'Tm-169*',
+            'Yb': 'Yb-174',
+            'Lu': 'Lu-175',
+            'Hf': 'Hf-180',
+            'Ta': 'Ta-181',
+            'W': 'W-184',
+            'Re': 'Re-187',
+            'Os': 'Os-192',
+            'Ir': 'Ir-193',
+            'Pt': 'Pt-195',
+            'Au': 'Au-197*',
+            'Hg': 'Hg-202',
+            'Tl': 'Tl-205',
+            'Pb': 'Pb-208',
+            'Bi': 'Bi-209*',
+            'Po': 'Po-209',
+            'At': 'At-210',
+            'Rn': 'Rn-222',
+            'Fr': 'Fr-223',
+            'Ra': 'Ra-226',
+            'Ac': 'Ac-227',
+            'Th': 'Th-232*',
+            'Pa': 'Pa-231*',
+            'U': 'U-238',
+            'Np': 'Np-237',
+            'Pu': 'Pu-244',
+            'Am': 'Am-243',
+            'Cm': 'Cm-247',
+            'Bk': 'Bk-247',
+            'Cf': 'Cf-251',
+            'Es': 'Es-252',
+            'Fm': 'Fm-257',
+            'Md': 'Md-258',
+            'No': 'No-259',
+            'Lr': 'Lr-262',
+            'Rf': 'Rf-267',
+            'Db': 'Db-268',
+            'Sg': 'Sg-269',
+            'Bh': 'Bh-270',
+            'Hs': 'Hs-270',
+            'Mt': 'Mt-278',
+            'Ds': 'Ds-281',
+            'Rg': 'Rg-282',
+            'Cn': 'Cn-285',
+            'Nh': 'Nh-286',
+            'Fl': 'Fl-289',
+            'Mc': 'Mc-290',
+            'Lv': 'Lv-293',
+            'Ts': 'Ts-294',
+            'Og': 'Og-294'
+}
+
