--- CMakeLists.txt.orig	2023-07-05 10:28:35.568953194 +0100
+++ CMakeLists.txt	2023-07-05 10:29:46.095673975 +0100
@@ -13,14 +13,15 @@
 
 # Development vs. Development.Module
 # https://cmake.org/cmake/help/latest/module/FindPython3.html?highlight=Development.Module
-if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0" AND NOT APPLE)
-    set(DEVELOPMENT_COMPONENT "Development.Module")
-else()
-    set(DEVELOPMENT_COMPONENT "Development")
-endif()
+#if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0" AND NOT APPLE)
+#    set(DEVELOPMENT_COMPONENT "Development.Module")
+#else()
+#    set(DEVELOPMENT_COMPONENT "Development")
+#endif()
 
 # find Python3
-find_package(Python3 COMPONENTS Interpreter ${DEVELOPMENT_COMPONENT} NumPy REQUIRED)
+#find_package(Python3 COMPONENTS Interpreter ${DEVELOPMENT_COMPONENT} NumPy REQUIRED)
+set(Python3_NumPy_INCLUDE_DIRS "$ENV{EBROOTSCIPYMINBUNDLE}/lib/python3.10/site-packages/numpy/core/include/numpy")
 
 # find PDAL. Require 2.1+
 find_package(PDAL 2.4 REQUIRED)
