--- orig/CMakeLists.txt	2023-06-30 14:31:31.103305630 +0100
+++ CMakeLists.txt	2023-06-30 14:32:27.293695326 +0100
@@ -6,18 +6,20 @@
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_BUILD_TYPE "Release")
 
+# This didn't appear to work with 3.23.1
 # Python-finding settings
-set(Python3_FIND_STRATEGY "LOCATION")
-set(Python3_FIND_REGISTRY "LAST")
-set(Python3_FIND_FRAMEWORK "LAST")
+#set(Python3_FIND_STRATEGY "LOCATION")
+#set(Python3_FIND_REGISTRY "LAST")
+#set(Python3_FIND_FRAMEWORK "LAST")
 
+# This didn't appear to work with 3.23.1
 # Development vs. Development.Module
 # https://cmake.org/cmake/help/latest/module/FindPython3.html?highlight=Development.Module
-if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0" AND NOT APPLE)
-    set(DEVELOPMENT_COMPONENT "Development.Module")
-else()
-    set(DEVELOPMENT_COMPONENT "Development")
-endif()
+#if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0" AND NOT APPLE)
+#    set(DEVELOPMENT_COMPONENT "Development.Module")
+#else()
+#    set(DEVELOPMENT_COMPONENT "Development")
+#endif()
 
 # find Python3
 find_package(Python3 COMPONENTS Interpreter ${DEVELOPMENT_COMPONENT} NumPy REQUIRED)
