--- /dev/shm/yearwoog/build-2021b-21b/tree/2.0.4/system-system/tree-2.0.4/Makefile	2022-09-06 14:31:53.000000000 +0100
+++ Makefile	2022-11-01 10:54:12.765866000 +0000
@@ -16,7 +16,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
-PREFIX=/usr/local
+#PREFIX=/usr/local
 
 CC=gcc
 INSTALL=install
@@ -106,7 +106,7 @@
 	$(INSTALL) -d $(DESTDIR)
 	$(INSTALL) -d $(MANDIR)/man1
 	$(INSTALL) $(TREE_DEST) $(DESTDIR)/$(TREE_DEST); \
-	$(INSTALL) -m 644 doc/$(MAN) $(MANDIR)/man1/$(MAN)
+	#$(INSTALL) -m 644 doc/$(MAN) $(MANDIR)/man1/$(MAN)
 
 distclean:
 	rm -f *.o *~
