diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50fdcfc..5a4f458 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,8 +87,9 @@ ExternalProject_Add(rocksdb
     URL https://github.com/facebook/rocksdb/archive/v6.22.1.tar.gz
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/external
     CONFIGURE_COMMAND ""
+    PATCH_COMMAND patch -p1 -i ${CMAKE_BINARY_DIR}/../GLnexus-1.4.1_rocksdb.patch 
     BUILD_IN_SOURCE 1
-    BUILD_COMMAND bash -c "export PORTABLE=1 && export DISABLE_JEMALLOC=1 && export OPT='-DNDEBUG -O3 -march=ivybridge -msse4.2 -DHAVE_SSE42 -mpclmul -DROCKSDB_NO_DYNAMIC_EXTENSION' && make -n static_lib && make -j$(nproc) static_lib"
+    BUILD_COMMAND bash -c "export PORTABLE=1 && export DISABLE_JEMALLOC=1 && export OPT='-DNDEBUG -O3 -march=ivybridge -msse4.2 -DHAVE_SSE42 -mpclmul -mavx2 -DROCKSDB_NO_DYNAMIC_EXTENSION -DWITH_LZ4=OFF -ULZ4' && make -n static_lib && make -j$(nproc) static_lib"
     INSTALL_COMMAND ""
     LOG_DOWNLOAD ON
     LOG_BUILD ON
@@ -165,6 +166,7 @@ include_directories(${SPDLOG_INCLUDE_DIR})
 ExternalProject_Add(capnp
     URL https://capnproto.org/capnproto-c++-0.7.0.tar.gz
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/external
+    PATCH_COMMAND patch -p2 -i ${CMAKE_BINARY_DIR}/../GLnexus-1.4.1_gcc13-uint8.patch && patch -p2 -i ${CMAKE_BINARY_DIR}/../GLnexus-1.4.1_test-skipfailed.patch
     CONFIGURE_COMMAND ./configure --prefix=${CMAKE_BINARY_DIR}/external
     BUILD_IN_SOURCE 1
     BUILD_COMMAND bash -c "make -j$(nproc) check"
@@ -183,18 +185,18 @@ set(CMAKE_CXX_FLAGS_RELEASE "-gdwarf -DNDEBUG -O3")
 set(COMMON_INCLUDES ${PROJECT_SOURCE_DIR}/include)
 include_directories(${COMMON_INCLUDES} src)
 include_directories(${COMMON_INCLUDES} src/rocksdb/include)
-include_directories(${COMMON_INCLUDES} capnp/serialize)
+include_directories(${COMMON_INCLUDES} ${EBBUILDGLNEXUS}/capnp/serialize)

 # Invoke cap'n proto IDL compiler, to generate C++ serialization code from
 # definitions.
 add_custom_command(
-  OUTPUT  capnp/serialize/defs.capnp.c++ capnp/serialize/defs.capnp.h
-  COMMAND ${CMAKE_BINARY_DIR}/external/bin/capnp compile -I${CMAKE_CURRENT_BINARY_DIR}/external/include -oexternal/bin/capnpc-c++ capnp/serialize/defs.capnp
+  OUTPUT  ${EBBUILDGLNEXUS}/capnp/serialize/defs.capnp.c++ ${EBBUILDGLNEXUS}/capnp/serialize/defs.capnp.h
+  COMMAND ${CMAKE_BINARY_DIR}/external/bin/capnp compile -I${CMAKE_CURRENT_BINARY_DIR}/external/include -oexternal/bin/capnpc-c++ --src-prefix=${EBBUILDGLNEXUS} ${EBBUILDGLNEXUS}/capnp/serialize/defs.capnp
   DEPENDS capnp/serialize/defs.capnp
   COMMENT "GLnexus: generating cap'n proto C++ and H files")

 add_library(glnexus
-            capnp/serialize/defs.capnp.h capnp/serialize/defs.capnp.c++
+            ${EBBUILDGLNEXUS}/capnp/serialize/defs.capnp.h ${EBBUILDGLNEXUS}/capnp/serialize/defs.capnp.c++
             include/types.h src/types.cc
             include/data.h src/data.cc
             include/compare_queries.h src/compare_queries.cc
