--- ../CMakeLists.txt	2024-06-19 11:42:27.788354153 +0100
+++ CMakeLists.txt	2024-06-19 11:46:01.046729364 +0100
@@ -372,10 +372,10 @@
 
 set(METALIBS ${METALIBS}
   -lKMC
-  -lhts zlib
+  -lhts -lz
   -lsdsl
   brwt
-  jsoncpp_lib
+  -ljsoncpp
   progress_bar
   mersenne_twister
   spdlog::spdlog
@@ -602,6 +602,6 @@
 
   SET_SOURCE_FILES_PROPERTIES(${interface_files} PROPERTIES CPLUSPLUS ON)
 
-  SWIG_ADD_LIBRARY(metagraph LANGUAGE python SOURCES ${interface_files})
-  SWIG_LINK_LIBRARIES(metagraph metagraph-core ${METALIBS} ${PYTHON_LIBRARIES})
+  SWIG_ADD_LIBRARY(metagraph-py LANGUAGE python SOURCES ${interface_files})
+  SWIG_LINK_LIBRARIES(metagraph-py metagraph-core ${METALIBS} ${PYTHON_LIBRARIES})
 endif()
