diff --git a/pyproject.toml b/pyproject.toml
index 324447b6..91387a75 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -15,8 +15,7 @@ maintainers = [
 description = 'MacSyFinder: Detection of macromolecular systems in protein datasets using systems modelling and similarity search'
 keywords = ['science', 'research', 'bioinformatic', 'microbiology', 'genomic', 'prokaryote', 'functional gene annotation']
 readme = "README.md"
-license = "GPL-3.0-or-later"
-license-files = ["COPYING"]
+license = {file = 'COPYING'}
 
 classifiers = [
     "Programming Language :: Python :: 3",
