--- src/Makefile	2021-07-14 10:20:20.137672000 +0100
+++ src/Makefile	2021-07-14 13:34:28.348374000 +0100
@@ -4,19 +4,19 @@

-override LDLIBS += -lgsl -lopenblas -lm -lnick
+override LDLIBS += -lgsl -lflexiblas -lm -lnick
 # Some Linux distributions require separate lapacke library
-# override LDLIBS += -llapacke
+# override LDLIBS += -lscalapack
 override LDFLAGS += -g  -p -pg -L./nicksrc
-override CFLAGS += -c -g -p -pg -Wimplicit -I./ -I./nicksrc -fno-common
+override CFLAGS += -c -g -p -pg -Wimplicit -I./ -I./nicksrc -I$EBROOTFLEXIBLAS/include -I$EBROOTGSL/include
 # Mac additions using homebrew installations
 #override CFLAGS += -I/usr/local/opt/openblas/include -I/usr/local/opt/gsl/include
 #override LDFLAGS += -L/usr/local/opt/openblas/lib -L/usr/local/opt/gsl/lib
 # Harvard Medical School O2 cluster additions
-ifdef SLURM_CONF
-override CFLAGS += -I/n/app/openblas/0.2.19/include -I/n/app/gsl/2.3/include
-override LDFLAGS += -L/n/app/openblas/0.2.19/lib -L/n/app/gsl/2.3/lib/
-TOP=../bin
-BIN=$(HOMEL)/../bin
-endif
+#ifdef SLURM_CONF
+#override CFLAGS += -I/n/app/openblas/0.2.19/include -I/n/app/gsl/2.3/include
+#override LDFLAGS += -L/n/app/openblas/0.2.19/lib -L/n/app/gsl/2.3/lib/
+#TOP=../bin
+#BIN=$(HOMEL)/../bin
+#endif
 
 ND = nicksrc
 NLIB = $(ND)/libnick.a
