--- Makefile.orig	2024-02-05 16:48:49.879689090 +0000
+++ Makefile	2024-02-05 16:58:15.702425689 +0000
@@ -3,23 +3,23 @@
 # Set the following variables before building:
 # Path to original CalculiX source (e.g. $(HOME)/ccx_2.xx/src )
 CCX_VERSION			= 2.20
-CCX             = $(HOME)/CalculiX/ccx_$(CCX_VERSION)/src
+CCX             = ebbuilddirgrep/CalculiX/ccx_$(CCX_VERSION)/src
 
 ### Change these if you built SPOOLES, ARPACK, or yaml-cpp from source ###
 # SPOOLES include flags (e.g. -I$(HOME)/SPOOLES.2.2 )
-SPOOLES_INCLUDE   = -I/usr/include/spooles/
+SPOOLES_INCLUDE   = -I${EBROOTSPOOLES}/include/spooles
 # SPOOLES library flags (e.g. $(HOME)/SPOOLES.2.2/spooles.a)
-SPOOLES_LIBS      = -lspooles
+SPOOLES_LIBS      = -L${EBROOTSPOOLES}/lib -lspooles
 #
 # ARPACK include flags (e.g. -I$(HOME)/ARPACK)
-ARPACK_INCLUDE    =
+ARPACK_INCLUDE    = -I${EBROOTARPACKMINNG}/include/arpack
 # ARPACK library flags (e.g. $(HOME)/ARPACK/libarpack_INTEL.a)
-ARPACK_LIBS       = -larpack -llapack -lblas
+ARPACK_LIBS       = -L${EBROOTARPACKMINNG}/lib -L${EBROOTFLEXIBLAS}/lib -larpack -lflexiblas
 #
 # yaml-cpp include flags (e.g. -I$(HOME)/yaml-cpp/include)
-YAML_INCLUDE      = -I/usr/include/
+YAML_INCLUDE      = -I${EBROOTYAMLMINCPP}/include/yaml-cpp/
 # yaml-cpp library flags (e.g. -L$(HOME)/yaml-cpp/build -lyaml-cpp)
-YAML_LIBS         = -lyaml-cpp
+YAML_LIBS         = -L${EBROOTYAMLMINCPP}/lib -lyaml-cpp
 
 # Get the CFLAGS and LIBS from pkg-config (preCICE version >= 1.4.0).
 # If pkg-config cannot find the libprecice.pc meta-file, you may need to set the
@@ -62,7 +62,7 @@
 	CC = mpicc
 endif
 
-FFLAGS = -Wall -O3 -fopenmp $(INCLUDES) ${ADDITIONAL_FFLAGS}
+FFLAGS = -Wall -O3 -fopenmp $(INCLUDES) ${ADDITIONAL_FFLAGS} -fallow-argument-mismatch
 # Note for GCC 10 or newer: add -fallow-argument-mismatch in the above flags
 FC = mpifort
 # FC = mpif90
