diff -Nur 2022R2.orig/bear/Framework/bin/Linux64/ssh 2022R2/bear/Framework/bin/Linux64/ssh
--- 2022R2.orig/bear/Framework/bin/Linux64/ssh	1970-01-01 01:00:00.000000000 +0100
+++ 2022R2/bear/Framework/bin/Linux64/ssh	2021-04-27 16:18:05.550124000 +0100
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# BEAR wrapper for ssh, to avoid symbol conflicts with libraries shipped within ANSYS
+
+set -eux
+
+unset LD_LIBRARY_PATH
+
+exec /usr/bin/ssh $@
diff -Nur 2022R2.orig/bear/licensingclient/linx64/ansyscl.bear 2022R2/bear/licensingclient/linx64/ansyscl.bear
--- 2022R2.orig/bear/licensingclient/linx64/ansyscl.bear	1970-01-01 01:00:00.000000000 +0100
+++ 2022R2/bear/licensingclient/linx64/ansyscl.bear	2021-04-27 16:19:56.972275000 +0100
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# ansyscl wrapper by BEAR to avoid missing symbol errors.
+# We link libz.so to libldap-2.4.so.2 in this directory, thus stopping openssl from including the LDAP library, which
+# brings kerberos and with it the symbol errors. (See the easyconfig for details).
+
+set -eux
+
+HERE=$(dirname $0)
+
+export LD_LIBRARY_PATH="${HERE}:${LD_LIBRARY_PATH}"
+
+exec ${HERE}/ansyscl.bin $@
