--- CMakeLists.txt.orig	2023-07-26 14:42:15.962739126 +0100
+++ CMakeLists.txt	2023-07-26 14:43:47.174992539 +0100
@@ -60,8 +60,10 @@
 # Find MPI
 find_package(MPI REQUIRED COMPONENTS C CXX Fortran)

+# change this to use set so ; delimiter isn't appended. string(APPEND... also possible
 if("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU")
-  list(APPEND CMAKE_Fortran_FLAGS "-ffree-line-length-none")
+# list(APPEND CMAKE_Fortran_FLAGS "-ffree-line-length-none")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-line-length-none")
 endif()

 set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/mod)
