diff -Nru DualSPHysics-5.0.233.orig/src/source/Makefile DualSPHysics-5.0.233/src/source/Makefile
--- DualSPHysics-5.0.233.orig/src/source/Makefile	2023-07-13 23:36:10.404303000 +0100
+++ DualSPHysics-5.0.233/src/source/Makefile	2023-07-14 10:13:15.298651000 +0100
@@ -4,7 +4,7 @@
 USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -55,53 +55,12 @@
   CCFLAGS:=$(CCFLAGS) -DDISABLE_MOORDYN
 endif
 
-#=============== CUDA selection ===============
-CUDAVER=92
-
 #=============== CUDA toolkit directory (make appropriate for local CUDA installation) ===============
-ifeq ($(CUDAVER),00)
-  DIRTOOLKIT=/usr/local/cuda
-endif
-ifeq ($(CUDAVER),75)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-7.5
-endif
-ifeq ($(CUDAVER),91)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.1
-endif
-ifeq ($(CUDAVER),92)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.2
-endif
+DIRTOOLKIT=$(EBROOTCUDA)
 
 #=============== Select GPU architectures ===============
-ifeq ($(CUDAVER),00)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_20,code=\"sm_20,compute_20\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-endif
-ifeq ($(CUDAVER),75)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_20,code=\"sm_20,compute_20\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_37,code=\"sm_37,compute_37\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-endif
-ifeq ($(CUDAVER),91)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
-endif
-ifeq ($(CUDAVER),92)
-  # module load cuda/9.2
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
-endif
+GENCODE:=$(GENCODE) -gencode=arch=compute_60,code=\"sm_60,compute_60\"
+GENCODE:=$(GENCODE) -gencode=arch=compute_80,code=\"sm_80,compute_80\"
 
 
 #=============== Files to compile ===============
diff -Nru DualSPHysics-5.0.233.orig/src/source/Makefile_cpu DualSPHysics-5.0.233/src/source/Makefile_cpu
--- DualSPHysics-5.0.233.orig/src/source/Makefile_cpu	2023-07-13 23:36:10.405708181 +0100
+++ DualSPHysics-5.0.233/src/source/Makefile_cpu	2023-07-13 23:37:46.419298390 +0100
@@ -4,7 +4,7 @@
 USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
diff -Nru DualSPHysics-5.0.233.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile DualSPHysics-5.0.233/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile
--- DualSPHysics-5.0.233.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile	2023-07-13 23:36:11.729366000 +0100
+++ DualSPHysics-5.0.233/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile	2023-07-14 10:14:54.340197000 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics NNewtonian GPU/CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -15,7 +15,7 @@
 LIBS_DIRECTORIES:=$(LIBS_DIRECTORIES) -L../lib/linux_gcc
 
 EXECNAME=DualSPHysics5.0_NNewtonian_linux64
-EXECS_DIRECTORY=../../../bin/linux/DSNNewtonian
+EXECS_DIRECTORY=../../../bin/linux
 
 # -std=c++0x ---> Used to avoid errors for calls to enums
 ifeq ($(USE_DEBUG), YES)
@@ -55,53 +55,12 @@
   CCFLAGS:=$(CCFLAGS) -DDISABLE_MOORDYN
 endif
 
-#=============== CUDA selection ===============
-CUDAVER=92
-
 #=============== CUDA toolkit directory (make appropriate for local CUDA installation) ===============
-ifeq ($(CUDAVER),00)
-  DIRTOOLKIT=/usr/local/cuda
-endif
-ifeq ($(CUDAVER),75)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-7.5
-endif
-ifeq ($(CUDAVER),91)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.1
-endif
-ifeq ($(CUDAVER),92)
-  DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.2
-endif
+DIRTOOLKIT=$(EBROOTCUDA)
 
 #=============== Select GPU architectures ===============
-ifeq ($(CUDAVER),00)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_20,code=\"sm_20,compute_20\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-endif
-ifeq ($(CUDAVER),75)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_20,code=\"sm_20,compute_20\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_37,code=\"sm_37,compute_37\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-endif
-ifeq ($(CUDAVER),91)
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
-endif
-ifeq ($(CUDAVER),92)
-  # module load cuda/9.2
-  GENCODE:=$(GENCODE) -gencode=arch=compute_30,code=\"sm_30,compute_30\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_35,code=\"sm_35,compute_35\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_50,code=\"sm_50,compute_50\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
-  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
-endif
+GENCODE:=$(GENCODE) -gencode=arch=compute_60,code=\"sm_60,compute_60\"
+GENCODE:=$(GENCODE) -gencode=arch=compute_80,code=\"sm_80,compute_80\"
 
 
 #=============== Files to compile ===============
diff -Nru DualSPHysics-5.0.233.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu DualSPHysics-5.0.233/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu
--- DualSPHysics-5.0.233.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu	2023-07-13 23:36:11.731310855 +0100
+++ DualSPHysics-5.0.233/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu	2023-07-14 09:45:26.879917402 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics NNewtonian CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -15,7 +15,7 @@
 LIBS_DIRECTORIES:=$(LIBS_DIRECTORIES) -L../lib/linux_gcc
 
 EXECNAME=DualSPHysics5.0_NNewtonianCPU_linux64
-EXECS_DIRECTORY=../../../bin/linux/DSNNewtonian
+EXECS_DIRECTORY=../../../bin/linux
 
 # -std=c++0x ---> Used to avoid errors for calls to enums
 ifeq ($(USE_DEBUG), YES)
