diff --git a/src/SGA/rmdup.cpp b/src/SGA/rmdup.cpp
index 425ddb2..731c740 100644
--- a/src/SGA/rmdup.cpp
+++ b/src/SGA/rmdup.cpp
@@ -232,7 +232,7 @@ std::string parseDupHits(const StringVector& hitsFilenames, const std::string& o
     while(!done)
     {
         // Parse a line from the current file
-        bool valid = getline(*reader_vec[currReaderIdx], line);
+        bool valid = static_cast<bool>(getline(*reader_vec[currReaderIdx], line));
         ++numRead;
         // Deal with switching the active reader and the end of files
         if(!valid || numRead == buffer_size)
diff --git a/src/Util/ClusterReader.cpp b/src/Util/ClusterReader.cpp
index 5c9460b..6b626e3 100644
--- a/src/Util/ClusterReader.cpp
+++ b/src/Util/ClusterReader.cpp
@@ -67,7 +67,7 @@ bool ClusterReader::generate(ClusterVector& out)
 bool ClusterReader::readCluster(ClusterRecord& record)
 {
     std::string line;
-    bool good = getline(*m_pReader, line);
+    bool good = static_cast<bool>(getline(*m_pReader, line));
     if(!good || line.empty())
         return false;
     std::stringstream parser(line);
diff --git a/src/Util/StdAlnTools.cpp b/src/Util/StdAlnTools.cpp
index b6441eb..237ce9b 100644
--- a/src/Util/StdAlnTools.cpp
+++ b/src/Util/StdAlnTools.cpp
@@ -119,7 +119,7 @@ std::string StdAlnTools::expandCigar(const std::string& cigar)
     char code;
     while(parser >> length)
     {
-        bool success = parser >> code;
+        bool success{ (parser >> code).good() };
         expanded.append(length, code);
         assert(success);
         (void)success;
diff --git a/src/bin/sga-align b/src/bin/sga-align
old mode 100755
new mode 100644
diff --git a/src/bin/sga-astat.py b/src/bin/sga-astat.py
old mode 100755
new mode 100644
diff --git a/src/bin/sga-bam2de.pl b/src/bin/sga-bam2de.pl
old mode 100755
new mode 100644
diff --git a/src/configure.ac b/src/configure.ac
index cf6c789..530727b 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -92,7 +92,7 @@ fi
 
 # Set compiler flags.
 AC_SUBST(AM_CXXFLAGS, "-Wall -Wextra $fail_on_warning -Wno-unknown-pragmas")
-AC_SUBST(CXXFLAGS, "-std=c++98 -O3")
+AC_SUBST(CXXFLAGS, "-std=c++11 -O3")
 AC_SUBST(CFLAGS, "-std=gnu99 -O3")
 AC_SUBST(CPPFLAGS, "$CPPFLAGS $openmp_cppflags $sparsehash_include $bamtools_include")
 AC_SUBST(LDFLAGS, "$openmp_cppflags $external_malloc_ldflags $bamtools_ldflags $LDFLAGS")
